internal/serviceapi/resourcepolicyapi/resource_schema.go

```diff
@@ -17,6 +18,7 @@ func ResourceSchema(ctx context.Context) schema.Schema {
 			"created_by_user": schema.SingleNestedAttribute{
 				Computed:            true,
 				MarkdownDescription: "The user that last updated the atlas resource policy.",
+				CustomType:          customtype.NewObjectType[TFCreatedByUserModel](ctx),
```

Comment:
question for my understanding: Currently we leverage the type defined in the TFModel (e.g. customtype.ObjectValue[TFCreatedByUserModel]) during conversion logic. Why is the custom type required in the schema definition? Is there some form of error returned by TPF due to mismatch of TFModel with TF schema?